core/test-runner.ts

```diff
@@ -18,6 +19,16 @@ export class TestRunner {
    private _testResults = new TestSetResults();
    private _currentTestResults: TestResults;
    private _currentTestFixtureResults: TestFixtureResults;
+   private _output: TestOutput;
+
+   constructor (output?: TestOutput) {
+       // If we were given a TestOutput, use it, otherwise make one
+       if (output !== undefined) {
```

Comment:
This is interesting! Is this to allow the current empty constructor to be used as is? A way I've used to resolve the dependencies in a similar way is using a factory function, not sure which is better to be honest, what were your thoughts?